cylc/flow/rundb.py

```diff
@@ -487,78 +455,42 @@ def _execute_stmt(self, stmt, stmt_args_list):
             LOG.warning(err_log)
             raise
 
-    def pre_select_broadcast_states(self, id_key=None, order=None):
+    def pre_select_broadcast_states(self, order=None):
         """Query statement and args formation for select_broadcast_states."""
         form_stmt = r"SELECT point,namespace,key,value FROM %s"
         if order == "ASC":
             ordering = " ORDER BY point ASC, namespace ASC, key ASC"
             form_stmt = form_stmt + ordering
-        if id_key is None or id_key == self.CHECKPOINT_LATEST_ID:
-            return form_stmt % self.TABLE_BROADCAST_STATES, []
-        else:
-            return (form_stmt % self.TABLE_BROADCAST_STATES_CHECKPOINTS +
-                    r" WHERE id==?"), [id_key]
+        return form_stmt % self.TABLE_BROADCAST_STATES
 
-    def select_broadcast_states(self, callback, id_key=None, sort=None):
-        """Select from broadcast_states or broadcast_states_checkpoints.
+    def select_broadcast_states(self, callback, sort=None):
+        """Select from broadcast_states.
 
         Invoke callback(row_idx, row) on each row, where each row contains:
             [point, namespace, key, value]
-
-        If id_key is specified,
-        select from broadcast_states table if id_key == CHECKPOINT_LATEST_ID.
-        Otherwise select from broadcast_states_checkpoints where id == id_key.
         """
-        stmt, stmt_args = self.pre_select_broadcast_states(id_key=id_key,
-                                                           order=sort)
-        for row_idx, row in enumerate(self.connect().execute(stmt, stmt_args)):
+        stmt = self.pre_select_broadcast_states(order=sort)
+        for row_idx, row in enumerate(self.connect().execute(stmt)):
             callback(row_idx, list(row))
 
-    def select_checkpoint_id(self, callback, id_key=None):
-        """Select from checkpoint_id.
+    def select_suite_params(self, callback):
+        """Select from suite_params.
 
         Invoke callback(row_idx, row) on each row, where each row contains:
-            [id, time, event]
-
-        If id_key is specified, add where id == id_key to select.
+            [key,value]
```

Comment:
Can you be more explicit about the data the key and value represent?

Reply:
The diff shows up weirdly; I didn't actually change this part of the docstring at all:
```
    def select_suite_params(self, callback):
        """Select from suite_params.

        Invoke callback(row_idx, row) on each row, where each row contains:
            [key,value]
```

Reply:
Added an example of a row: https://github.com/cylc/cylc-flow/pull/3906/commits/6c07d9fd777fe67940f7c06aab2ac221dae6b456